### DISCUSSION TITLE
SailorHat with ESP as NMEA0183 gateway

### DISCUSSION THREAD

**Discussion by jonasbager** [hatlabs/discussions#54] [Q&A]
Hi. Is it possible to attach NMEA0183 devices (for example an Autopilot that only has 0183) to the Sailorhat and thus getting that device on my regular NMEA2000 network to be controlled by the NMEA2000 plotter?

**Comment by mairas** [comment]
You need to use an RS422 or an RS485 module or two (search for "isolated rs485 module" on Aliexpress) to interface NMEA0183 with an SH-ESP32. Then you need to write software that translates the MFD N2K sentences into NMEA0183 sentences understood by the autopilot and vice versa.
